vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -464,52 +474,66 @@ && getElement().getNode().getParent() == null) {
                         }
                     });
         }
-        super.setOpened(opened);
+        getElement().setProperty("opened", opened);
     }
 
     /**
-     * <p>
-     * Description copied from corresponding location in WebComponent:
-     * </p>
-     * <p>
      * True if the notification is currently displayed.
      * <p>
-     * This property is synchronized automatically from client side when a
-     * 'opened-changed' event happens.
-     * </p>
+     * This property is synchronized automatically from client side when an
+     * {@code opened-changed} event happens.
      *
      * @return the {@code opened} property from the webcomponent
      */
+    @Synchronize(property = "opened", value = "opened-changed")
     public boolean isOpened() {
-        return isOpenedBoolean();
+        return getElement().getProperty("opened", false);
     }
 
-    @Override
+    /**
+     * {@code opened-changed} event is sent when the notification opened state
+     * changes.
+     */
+    public static class OpenedChangeEvent extends ComponentEvent<Notification> {
+        private final boolean opened;
+
+        public OpenedChangeEvent(Notification source, boolean fromClient) {
+            super(source, fromClient);
+            this.opened = source.isOpened();
+        }
+
+        public boolean isOpened() {
+            return opened;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code opened-changed} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
     public Registration addOpenedChangeListener(
-            ComponentEventListener<OpenedChangeEvent<Notification>> listener) {
-        return super.addOpenedChangeListener(listener);
+            ComponentEventListener<OpenedChangeEvent> listener) {
+        return getElement().addPropertyChangeListener("opened",
+                event -> listener.onComponentEvent(
+                        new OpenedChangeEvent(this, event.isUserOriginated())));
     }
 
     /**
-     * <p>
-     * Description copied from corresponding location in WebComponent:
-     * </p>
-     * <p>
      * The duration in milliseconds to show the notification. Set to {@code 0}
      * or a negative number to disable the notification auto-closing.
-     * </p>
      *
      * @param duration
      *            the value to set
      */
     public void setDuration(int duration) {
-        setDuration((double) duration);
+        getElement().setProperty("duration", (double) duration);
```

Comment:
```suggestion
        getElement().setProperty("duration", duration);
```
Nit: Cast is redundant, there is an overload for ints.